opengemini/statement_parser.go

```diff
@@ -0,0 +1,266 @@
+		return StatementTypeInsert
+	}
+
+	return StatementTypeUnknown
+}
+
+// isQueryKeyword checks if the keyword indicates a query statement
+func isQueryKeyword(word string) bool {
+	queryKeywords := []string{
+		"SELECT",
+		"SHOW",
+		"EXPLAIN",
+		"DESCRIBE",
+		"DESC",
+		"WITH",
+	}
+
+	for _, keyword := range queryKeywords {
+		if word == keyword {
+			return true
+		}
+	}
+	return false
+}
+
+// isCommandKeyword checks if the keyword indicates a command statement
+func isCommandKeyword(word string) bool {
+	commandKeywords := []string{
+		"CREATE",
+		"DROP",
+		"ALTER",
+		"UPDATE", // UPDATE RETENTION POLICY
+		"DELETE",
+	}
+
+	for _, keyword := range commandKeywords {
+		if word == keyword {
+			return true
+		}
+	}
+	return false
+}
+
+// isInsertKeyword checks if the keyword indicates an insert statement
+func isInsertKeyword(word string) bool {
+	return word == "INSERT"
+}
+
+// cleanCommand removes comments and extra whitespace
+func cleanCommand(command string) string {
+	trimmed := strings.TrimSpace(command)
+
+	// Handle single line comments (-- comment)
+	if idx := strings.Index(trimmed, "--"); idx != -1 {
+		trimmed = strings.TrimSpace(trimmed[:idx])
+	}
+
+	// Handle multiline comments (/* comment */)
+	for strings.Contains(trimmed, "/*") {
+		start := strings.Index(trimmed, "/*")
+		end := strings.Index(trimmed, "*/")
+		if start != -1 && end != -1 && end > start {
+			// Just remove the comment, don't add extra space
+			before := trimmed[:start]
+			after := trimmed[end+2:]
+
+			// If there's no space before comment and no space after comment,
+			// and both sides have content, add a space to separate words
+			needSpace := false
+			if len(before) > 0 && len(after) > 0 {
+				beforeEndsWithSpace := strings.HasSuffix(before, " ")
+				afterStartsWithSpace := strings.HasPrefix(after, " ")
+				if !beforeEndsWithSpace && !afterStartsWithSpace {
+					needSpace = true
+				}
+			}
+
+			if needSpace {
+				trimmed = before + " " + after
+			} else {
+				trimmed = before + after
+			}
+		} else {
+			break
+		}
+	}
+
+	return strings.TrimSpace(trimmed)
+}
+
+// parseInsertStatement parses INSERT statement into Point objects
+func parseInsertStatement(command string) ([]*Point, error) {
+	// Remove INSERT prefix from command
+	trimmed := strings.TrimSpace(command)
+	if !strings.HasPrefix(strings.ToUpper(trimmed), "INSERT") {
+		return nil, errors.New("not an INSERT statement")
+	}
+
+	// Extract Line Protocol part after INSERT keyword
+	// Format: INSERT measurement,tag1=value1 field1=value1,field2=value2
+	lpPart := strings.TrimSpace(trimmed[6:]) // Remove "INSERT"
+
+	// Parse Line Protocol to Point object
+	point, err := parseLineProtocolToPoint(lpPart)
+	if err != nil {
+		return nil, fmt.Errorf("invalid line protocol format: %w", err)
+	}
+
+	return []*Point{point}, nil
+}
+
+// parseLineProtocolToPoint converts line protocol string to Point
+func parseLineProtocolToPoint(lp string) (*Point, error) {
```

Comment:
The escape symbols in the protocol also need to be considered, line protocol parser reference https://github.com/openGemini/openGemini-cli/blob/ef4e06ca4042341fb7eca5c42521d38c01507173/core/line_protocol.go#L40